restbridge/RestBridge.cpp

```diff
@@ -41,38 +41,56 @@ class RestBridge
         }
     }
 
+    bool waitForRunningState(const uint16_t milliseconds)
+    {
+        sleep(milliseconds/1000.f);
+        return serverRunning_;
+    }
+
     void run( const std::string& schema )
     {
-        const std::string publisherSchema = schema + PUBLISHER_SCHEMA_SUFFIX;
-        const std::string subscriberSchema = schema + SUBSCRIBER_SCHEMA_SUFFIX;
-        RequestHandler handler( publisherSchema, subscriberSchema );
-
-        server::options options( handler );
-        std::stringstream port;
-        port << port_;
-        httpServer_.reset( new server( options.address( hostname_ ).port( port.str( ))));
-
-        LBINFO << "HTTP Server runnning on " << hostname_ << ":" << port_ << std::endl;
-        LBINFO << "ZeroEq Publisher Schema : " << publisherSchema << std::endl;
-        LBINFO << "ZeroEq Subscriber Schema: " << subscriberSchema << std::endl;
-        serverRunning_ = true;
-        httpServer_->run();
+        { // scope is for automic unlocking of the mutex_ when the http server is stopped
+            std::unique_lock<std::mutex> lock(mutex_);
+            const std::string publisherSchema = schema + PUBLISHER_SCHEMA_SUFFIX;
+            const std::string subscriberSchema = schema + SUBSCRIBER_SCHEMA_SUFFIX;
+            // Create request handler
+            RequestHandler handler( publisherSchema, subscriberSchema );
+
+            server::options options( handler );
+            std::stringstream port;
+            port << port_;
+            // Create http server
+            httpServer_.reset( new server( options.address( hostname_ ).port( port.str( ))));
+
+            std::cout << "HTTP Server runnning on " << hostname_ << ":" << port_ << std::endl;
```

Comment:
You still need some filtering for your messages. Now this means that the user of Livre is 'spammed' with this things, and I as user vote against that :)